x-pack/platform/plugins/shared/lens/public/editor_frame_service/editor_frame/config_panel/layer_panel.tsx

```diff
@@ -125,8 +152,8 @@ export function LayerPanel(props: LayerPanelProps) {
   };
 
   const datasourcePublicAPI = framePublicAPI.datasourceLayers?.[layerId];
-  const datasourceId = datasourcePublicAPI?.datasourceId;
-  let layerDatasourceState = datasourceId ? datasourceStates?.[datasourceId]?.state : undefined;
+  const datasourceId = datasourcePublicAPI?.datasourceId as unknown as 'formBased' | 'textBased';
```

Comment:
Why this change here?

Reply:
can datasourceId be other than formBased and textBased ?

Reply:
Oh datasourceId is a string?

Reply:
yes